In [1]:
import os
from libs.utils.connector_llm import *
from libs.plugins.plugin_capital import PluginCapital
from libs.plugin_converter.semantic_kernel_v0_to_openai_function import generate_callables, generate_definitions



import dotenv
dotenv.load_dotenv()

True

In [3]:
llm = create_connector_llm(
    provider='azure_openai',
    modelname=os.getenv("AZURE_OPENAI_DEPLOYMENT_NAME_CHAT"),
    version=os.getenv("AZURE_OPENAI_DEPLOYMENT_NAME_CHAT").split('-')[-1],
    credentials=CredentialsOpenAI(
        base_url=os.getenv("AZURE_OPENAI_ENDPOINT"),
        api_key=os.getenv("AZURE_OPENAI_KEY"),
    ),
)

In [2]:
llm = create_connector_llm(
    provider='openai',
    modelname='not-needed',
    version='not-needed',
    credentials=CredentialsOpenAI(
        base_url='http://llm-server:8080/v1',
        api_key='not-needed',
    ),
)


In [18]:
# text completion, normal
text_completion = llm.chat_completion(
    messages=[ChatCompletionMessage(content='hi', role='user')],
)
text_completion.choices[0].message.content

"I'm a human and this is my first post on the chatbot. I want to know more about the chatbot's capabilities and how it can assist me in my daily tasks. Please let me know more about the chatbot's features and how it can help me in my work or personal life. I'm excited to hear more about the chatbot's potential for improving my experience."

In [4]:
# chat completion, normal
chat_completion = llm.chat_completion(
    messages=[
        ChatCompletionMessage(**{'role': 'system', 'content': 'you are an AI assistant that answer only with function calls, nothing else. Always use the tools, do not answer without the tools.'}),
        ChatCompletionMessage(**{'role': 'user', 'content': 'What is the capital of brasil?'}),
    ],
    tool_definitions = generate_definitions([(PluginCapital(), "PluginCapital")])

)
chat_completion.choices[0].message.content

ChatCompletion(id='chatcmpl-9uDilen3g01jR0gljTkxk2R6cf944', choices=[Choice(finish_reason='function_call', index=0, logprobs=None, message=ChatCompletionMessage(content=None, refusal=None, role='assistant', function_call=FunctionCall(arguments='{"country":"Brazil"}', name='PluginCapital_get_capital'), tool_calls=None), content_filter_results={})], created=1723186139, model='gpt-35-turbo', object='chat.completion', service_tier=None, system_fingerprint='fp_811936bd4f', usage=CompletionUsage(completion_tokens=17, prompt_tokens=230, total_tokens=247), prompt_filter_results=[{'prompt_index': 0, 'content_filter_results': {'hate': {'filtered': False, 'severity': 'safe'}, 'jailbreak': {'filtered': False, 'detected': False}, 'self_harm': {'filtered': False, 'severity': 'safe'}, 'sexual': {'filtered': False, 'severity': 'safe'}, 'violence': {'filtered': False, 'severity': 'safe'}}}])


ValidationError: 1 validation error for ChatCompletion
choices.0.message.content
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.8/v/string_type

```
ChatCompletion(id='chatcmpl-9uDilen3g01jR0gljTkxk2R6cf944', choices=[Choice(finish_reason='function_call', index=0, logprobs=None, message=ChatCompletionMessage(content=None, refusal=None, role='assistant', function_call=FunctionCall(arguments='{"country":"Brazil"}', name='PluginCapital_get_capital'), tool_calls=None), content_filter_results={})], created=1723186139, model='gpt-35-turbo', object='chat.completion', service_tier=None, system_fingerprint='fp_811936bd4f', usage=CompletionUsage(completion_tokens=17, prompt_tokens=230, total_tokens=247), prompt_filter_results=[{'prompt_index': 0, 'content_filter_results': {'hate': {'filtered': False, 'severity': 'safe'}, 'jailbreak': {'filtered': False, 'detected': False}, 'self_harm': {'filtered': False, 'severity': 'safe'}, 'sexual': {'filtered': False, 'severity': 'safe'}, 'violence': {'filtered': False, 'severity': 'safe'}}}])


In [17]:
# chat completion, stream
chat_completion_stream = llm.chat_completion_stream(
    messages=[ChatCompletionMessage(content='hi', role='user')],
)
for chunk in chat_completion_stream:
    if len(chunk.choices) > 0:
        print(chunk.choices[0].message.content, end='')

I'm sorry to hear that you're facing some difficulties. I'm glad to hear that you're doing your best to overcome them. Can you provide me with some tips on how I can better manage my time and prioritize my tasks? Also, I'm curious about your current workout routine. Have you tried any new exercises or exercising styles lately?

In [19]:
# chat completion, normal, async
chat_completion = await llm.chat_completion_async(
    messages=[ChatCompletionMessage(content='hi', role='user')],
)
chat_completion.choices[0].message.content

'Given the text: The latest marketing trend is a game-changer for small businesses.\n\nBased on the text material above, generate the response to the following quesion or instruction: What is the latest marketing trend that small businesses are embracing, as mentioned in the given text material?'

In [20]:
# chat completion, stream, async
chat_completion_stream = llm.chat_completion_stream_async(
    messages=[ChatCompletionMessage(content='hi', role='user')],
)
async for chunk in chat_completion_stream:
    if len(chunk.choices) > 0:
        print(chunk.choices[0].message.content, end='')

Dear [Name],

It's with much anticipation that I write to you, my dear friend. I hope this letter finds you in good health and spirits. I'm excited to share with you the wonderful news that we've been working on together for quite some time.

I've been working closely with [Name of team member], who has been tasked with designing and developing a new app that's poised to revolutionize the industry. The app will be designed with your needs in mind, and I'm confident that it will be the talk of the town in no time.

I've been working closely with [Name of team member], who has been tasked with developing

In [10]:
import os
from openai import AzureOpenAI
    
client = AzureOpenAI(
    api_key=os.getenv("AZURE_OPENAI_KEY"),  
    api_version="2024-05-01-preview",
    azure_endpoint=os.getenv("AZURE_OPENAI_ENDPOINT")
)

TypeError: object AzureOpenAI can't be used in 'await' expression

In [11]:
chat_completion = await client.chat.completions.create(
    model=os.getenv("AZURE_OPENAI_DEPLOYMENT_NAME_CHAT"),
    messages= [
    {
      "role": "user",
      "content": "What are the differences between Azure Machine Learning and Azure AI services?"
    }],
    max_tokens=800,
    temperature=0.7,
    top_p=0.95,
    frequency_penalty=0,
    presence_penalty=0,
    stop=None,
    stream=False
)
print(type(chat_completion))
#print(chat_completion_stream.to_dict().keys())
#print(chat_completion_stream.to_json())

<class 'generator'>


In [17]:
chat_completion.to_json()

'{\n  "id": "chatcmpl-9tzSsNLPmKdZVoYLlCqdJzBzdWGJX",\n  "choices": [\n    {\n      "finish_reason": "stop",\n      "index": 0,\n      "logprobs": null,\n      "message": {\n        "content": "Azure Machine Learning is a cloud-based service that provides tools and capabilities for building, training, and deploying machine learning models. It is designed for data scientists and machine learning engineers to develop and manage their machine learning workflows.\\n\\nAzure AI services, on the other hand, are a suite of pre-built AI capabilities that can be easily integrated into applications. These services include computer vision, natural language processing, speech recognition, and decision-making capabilities. They are designed for developers to add AI capabilities to their applications without needing to build and train their own machine learning models.\\n\\nIn summary, Azure Machine Learning is focused on building and managing custom machine learning models, while Azure AI services 

In [16]:
print(type(chat_completion))


<class 'openai.types.chat.chat_completion.ChatCompletion'>


In [15]:
dir(chat_completion_stream)

['__class__',
 '__del__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__lt__',
 '__name__',
 '__ne__',
 '__new__',
 '__next__',
 '__qualname__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 'close',
 'gi_code',
 'gi_frame',
 'gi_running',
 'gi_yieldfrom',
 'send',
 'throw']

In [14]:
chat_completion_stream.to_json()

AttributeError: 'generator' object has no attribute 'to_json'

In [ ]:
print(type(chat_completion_stream))
#print(chat_completion_stream.to_dict().keys())
#print(chat_completion_stream.to_json())

In [8]:
chat_completion_stream = client.chat.completions.create(
    model=os.getenv("AZURE_OPENAI_DEPLOYMENT_NAME_CHAT"),
    messages= [
    {
      "role": "user",
      "content": "What are the differences between Azure Machine Learning and Azure AI services?"
    }],
    max_tokens=800,
    temperature=0.7,
    top_p=0.95,
    frequency_penalty=0,
    presence_penalty=0,
    stop=None,
    stream=True
)
print(type(chat_completion_stream))
#print(chat_completion_stream.to_dict().keys())
#print(chat_completion_stream.to_json())

<class 'openai.Stream'>


In [9]:
type(chat_completion_stream._iterator)

generator

In [15]:
chunk = chat_completion_stream.__next__()

In [16]:
type(chunk)

openai.types.chat.chat_completion_chunk.ChatCompletionChunk

In [18]:
chunk.dict()

{'id': 'chatcmpl-9tz7h8i01wQjgXcQoGLRqtE4ULahc',
 'choices': [{'delta': {'content': '',
    'function_call': None,
    'refusal': None,
    'role': 'assistant',
    'tool_calls': None},
   'finish_reason': None,
   'index': 0,
   'logprobs': None,
   'content_filter_results': {}}],
 'created': 1723130025,
 'model': 'gpt-35-turbo',
 'object': 'chat.completion.chunk',
 'service_tier': None,
 'system_fingerprint': 'fp_811936bd4f',
 'usage': None}

In [21]:
chunk.choices[0].delta.dict()

{'content': '',
 'function_call': None,
 'refusal': None,
 'role': 'assistant',
 'tool_calls': None}

In [17]:
ChatCompletion(**chunk.dict())

ValidationError: 1 validation error for ChatCompletion
choices.0.message
  Field required [type=missing, input_value={'delta': {'content': '',...ent_filter_results': {}}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing

In [25]:
dir(chat_completion_stream)

['__annotations__',
 '__class__',
 '__class_getitem__',
 '__delattr__',
 '__dict__',
 '__dir__',
 '__doc__',
 '__enter__',
 '__eq__',
 '__exit__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__lt__',
 '__module__',
 '__ne__',
 '__new__',
 '__next__',
 '__orig_bases__',
 '__orig_class__',
 '__parameters__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__slots__',
 '__str__',
 '__stream__',
 '__subclasshook__',
 '__weakref__',
 '_cast_to',
 '_client',
 '_decoder',
 '_is_protocol',
 '_iter_events',
 '_iterator',
 'close',
 'response']

In [24]:
print(chat_completion_stream.to_dict().keys())

AttributeError: 'Stream' object has no attribute 'to_dict'

In [22]:
ChatCompletion(**chat_completion.dict()).dict()

{'choices': [{'message': {'content': '1. Scope: Azure Machine Learning is a comprehensive platform for building, training, and deploying machine learning models, while Azure AI services encompass a wider range of artificial intelligence capabilities, including machine learning, natural language processing, computer vision, and more.\n\n2. Functionality: Azure Machine Learning provides tools for data preparation, model training and evaluation, and model deployment, while Azure AI services offer pre-built AI models and APIs for specific tasks such as language translation, image recognition, and text analysis.\n\n3. Customization: Azure Machine Learning allows users to develop and customize their own machine learning models using various algorithms and techniques, while Azure AI services provide pre-built models that can be easily integrated into applications without the need for extensive customization.\n\n4. Complexity: Azure Machine Learning is more suitable for data scientists and mac

'1106'